streams/src/main/java/org/apache/kafka/streams/processor/internals/TopologyMetadata.java

```diff
@@ -71,6 +73,7 @@
     private final Map<String, StateStore> globalStateStores = new HashMap<>();
     private final Set<String> allInputTopics = new HashSet<>();
     private final Map<String, Long> threadVersions = new ConcurrentHashMap<>();
+    private final ConcurrentMap<TaskId, ErrorTask> errorTasks = new ConcurrentHashMap<>();
```

Comment:
Not sure why we'd need to keep the error tasks mapping in two places and each is used to populate the other (see my other comment above). If the goal is to live beyond thread replacements then we don't we just keep it on the global level --- a.k.a. here, rather than within a thread's task executor as well? I.e. whenever we hit an exception we immediately call the topology metadata and add it here, so that we do not need the local mapping, and then for retrying the error tasks the thread would just try to loop over its assigned tasks' ids and see if any is in the mapping. This may be a bit less efficient but we avoid the complexity of bookkeeping in two places, a global mapping and a local per-thread mapping.